pageserver/src/tenant.rs

```diff
@@ -1699,11 +1728,30 @@ impl Tenant {
                 }
             }
         });
+
+        if stopping {
+            let timelines_accessor = self.timelines.lock().unwrap();
+            let not_broken_timelines = timelines_accessor
+                .values()
+                .filter(|timeline| timeline.current_state() != TimelineState::Broken);
+            for timeline in not_broken_timelines {
```

Comment:
could this be called from a map() method in the previous line ?

Reply:
With `.map` you'd have the problem of "iterators are lazy and do nothing yadda yadda". `.for_each(|tl| tl.set_state(TimelineState::Stopping))` exists for this reason.

Reply:
`not_broken_timelines` is an iterator and we need the precise number of elements it yielded (for `activated_timelines`).

So, leaning towards keeping the for loop.

Reply:
Funny how this thread is now here, explains why I failed to find it.

enumerate would just number each of the timelines with their index. Usually it feels icky to do modify state within `Iterator::map` and return the index, then force evaluation with `max` or leave the enumerate even and do a plain `fold`, so you can eventually get the count by `let count = not_broken_timelines....max().map(|index| index + 1).unwrap_or(0);`.

Luckily it seems no need for count anymore here.